test-lib-openshift.sh

```diff
@@ -1060,4 +1060,30 @@ function ct_os_check_cmd_internal() {
   return 1
 }
 
+# ct_os_test_image_stream
+# ------------------------
+# Creates an image stream and deploys a specified template. Then checks that a pod runs.
+# Argument: image_stream_file - local file name with an image stream
+# Argument: template_file - local file name with a template
+# Argument: service_name - how the pod will be named (prefix)
+# Argument: template_params (optional) - parameters for the template, like image stream version
+function ct_os_test_image_stream() {
+  local image_stream_file=$1
+  local template_file=$2
+  local service_name=$3
+  local template_params=${4:-}
+
+  echo "Running image stream test for stream $image_stream_file and template $template_file"
+  # shellcheck disable=SC2119
+  ct_os_new_project
+
+  oc create -f "${image_stream_file}"
+  # shellcheck disable=SC2086
+  oc process -p NAMESPACE="$(oc project -q)" ${template_params} -f "${template_file}" | oc create -f -
```

Comment:
What about if `oc process` failed. Would it be possible to add there check that `ct_os_wait_pod_ready` will be called only in success?

Reply:
By the way. This function is partially done by https://github.com/sclorg/container-common-scripts/blob/master/test-lib-openshift.sh#L275. What about to use it?

Reply:
Good catch, will call `ct_os_deploy_template_image` instead and also add the check there.

Reply:
Actually, for the file check itself, I believe the extra check is not necessary, because if the file does not exist, user will see pretty clear error message from the `oc process` command.